core/src/main/scala/com/microsoft/azure/synapse/ml/causal/LinearDMLEstimator.scala

```diff
@@ -0,0 +1,279 @@
+      3. Use the two models to fit a residual model on the second split.
+      4. Cross-fit by fitting the treatment and outcome models with the second split and the residual model with the first split.
+      5. Average slopes from the two residual models.
+    */
+    val splits = dataset.randomSplit(getSampleSplitRatio)
+    val (train, test) = (splits(0).cache, splits(1).cache)
+    val treatmentEffectDFV1 = setUpDMLPipeline(train, test)
+    val treatmentEffectDFV2 = setUpDMLPipeline(test, train)
+
+    // Average slopes from the two residual models.
+    val regressor = new GeneralizedLinearRegression()
+      .setLabelCol(SchemaConstants.OutcomeResidualColumn)
+      .setFeaturesCol(treatmentResidualVecCol)
+      .setFamily("gaussian")
+      .setLink("identity")
+      .setFitIntercept(false)
+    val (lrmV1, lrmV2) = (regressor.fit(treatmentEffectDFV1), regressor.fit(treatmentEffectDFV2))
+    Seq(train, test).foreach(_.unpersist)
+
+    val ate = Seq(lrmV1, lrmV2).map(_.coefficients(0)).sum / 2
+    ate
+  }
+
+  override def copy(extra: ParamMap): Estimator[LinearDMLModel] = {
+    defaultCopy(extra)
+  }
+
+  @DeveloperApi
+  override def transformSchema(schema: StructType): StructType = {
+    LinearDMLEstimator.validateTransformSchema(schema)
+  }
+}
+
+object LinearDMLEstimator extends ComplexParamsReadable[LinearDMLEstimator] {
+
+  def validateTransformSchema(schema: StructType): StructType = {
+    StructType(schema.fields)
+  }
+}
+
+/** Model produced by [[LinearDMLEstimator]]. */
+class LinearDMLModel(val uid: String)
+  extends Model[LinearDMLModel] with LinearDMLParams with ComplexParamsWritable with Wrappable with BasicLogging {
+  logClass()
+
+  def this() = this(Identifiable.randomUID("LinearDMLModel"))
+
+  var ates = new DoubleArrayParam(this, "ates", "treatment effect results for each iteration")
+  def getAtes: Array[Double] = $(ates)
+  def setAtes(v: Array[Double]): this.type = set(ates, v)
+
+  def getAte: Double = {
+    val finalAte =  $(ates).sum / $(ates).length
+    finalAte
+  }
+
+  def getCi: Array[Double] = {
+    val ciLowerBound = percentile($(ates), 100 * (1 - getConfidenceLevel))
+    val ciUpperBound = percentile($(ates), getConfidenceLevel * 100)
+    Array(ciLowerBound, ciUpperBound)
+  }
+
+  private def percentile(values: Seq[Double], quantile: Double): Double = {
+    val sortedValues = values.sorted
+    val percentile = new Percentile()
+    percentile.setData(sortedValues.toArray)
+    percentile.evaluate(quantile)
+  }
+
+  override def copy(extra: ParamMap): LinearDMLModel = defaultCopy(extra)
+
+  //scalastyle:off
+  /** LinearDMLEstimator transform does nothing by design and isn't supposed to be called by end user. */
+  override def transform(dataset: Dataset[_]): DataFrame = {
+    logTransform[DataFrame]({
+      dataset.toDF()
+    })
+  }
+
+  @DeveloperApi
+  override def transformSchema(schema: StructType): StructType =
```

Comment:
This transform schema doesent look right, you sure this doesent add any info to the data dframe?

Reply:
LinearDMLEstimator transform does nothing by design and isn't supposed to be called by end user.
Previously, I set it throw exception, but it won't pass fuzzing testing, so I changed it to return the original dataset back, in this case I don't think we need transform schema, please correct me if I missing something.

Reply:
So the linear DML Model should actually perform the regression. In particular you can think of this pipeline as estimating a prediction variable in two steps. The first is the debiasing operation where you map a dataframe to it's residuals. The second is the prediction of the target residuals. 

To form the actual prediction target, you first use your estimate of the effect from step 1, then add your predicted residual from step 2.

Reply:
To give a little more info here:

First use your learned residual models to map the inputs to their residuals, then use your treatment effect model to map the residuals to the treatment. Then append that treadment as the prediction column. (If im missing something here perhaps we can chat to help clarify)

Reply:
@memoryz , Jason, did you sync with our data scientist and if this is feasible?

Reply:
If there's no objections, I'll set it as by design.